gateway/src/apicast/policy/rate_limit/apicast-policy.json

```diff
@@ -0,0 +1,98 @@
+{
+  "$schema": "http://apicast.io/policy-v1/schema#manifest#",
+  "name": "rate limit policy",
+  "summary": "Adds rate limit.",
+  "description": ["This policy adds rate limit."],
+  "version": "builtin",
+  "configuration": {
+    "type": "object",
+    "properties": {
+      "limiters": {
+        "description": "List of limiters to be applied",
+        "type": "array",
+        "items": {
+          "oneOf": [{
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string",
+                "enum": ["connections"],
+                "description": "limiting request concurrency (or concurrent connections)"
+              },
+              "key": {
+                "description": "Key of limiter",
+                "type": "string"
+              },
+              "conn": {
+                "type": "integer",
+                "description": "the maximum number of concurrent requests allowed"
+              },
+              "burst": {
+                "type": "integer",
+                "description": "the number of excessive concurrent requests (or connections) allowed to be delayed"
+              },
+              "delay": {
+                "type": "number",
+                "description": "the default processing latency of a typical connection (or request)"
+              }
+            }
+          }, {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string",
+                "enum": ["leaky_bucket"],
+                "description": "limiting request rate"
+              },
+              "key": {
+                "description": "Key of limiter",
+                "type": "string"
+              },
+              "rate": {
+                "type": "integer",
+                "description": "the specified request rate (number per second) threshold"
+              },
+              "burst": {
+                "type": "integer",
+                "description": "the number of excessive requests per second allowed to be delayed"
+              }
+            }
+          }, {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string",
+                "enum": ["fixed_window"],
+                "description": "limiting request counts"
+              },
+              "key": {
```

Comment:
When we allow this to be a "template" in the future we will somehow have to distinguish between plain strings and templated strings.

We probably should change this to be more extensible in the future before the final release.

Also right now those keys have global scopes. That does not really work in our Cloud Hosted environment. We probably should offer a way to have "scope" of the key that could be either "global" or "service" and the key would be automatically namespaced.